src/rust/lib/flexer/src/group/rule.rs

```diff
@@ -0,0 +1,32 @@
+use crate::automata::pattern::Pattern;
+
+
+
+/// A rule is a pair of regex pattern and callback.
+/// The intention is to run the callback after encountering given pattern.
+#[derive(Clone,Debug)]
+pub struct Rule {
+    /// Pattern that triggers the callback.
+    pub pattern  : Pattern,
+    /// Callback containing stringified rust code.
+    pub callback : String,
+}
+
+/// Builder that allows us to add Rules to Group in a nice way.
```

Comment:
if you want to mentionm soemthig from the scope in docs, do not write it like here, use backticks around the names please! This way the generated docs actually will link to it.